applications/GeoMechanicsApplication/custom_utilities/math_utilities.hpp

```diff
@@ -633,474 +582,411 @@ class GeoMechanicsMathUtilities
-        Matrix(2,5) = 2.0*Tensor[78];
-
-        Matrix(3,0) = Tensor[9];
-        Matrix(3,1) = Tensor[13];
-        Matrix(3,2) = Tensor[18];
-        Matrix(3,3) = 2.0*Tensor[10];
-        Matrix(3,4) = 2.0*Tensor[14];
-        Matrix(3,5) = 2.0*Tensor[15];
-
-        Matrix(4,0) = Tensor[45];
-        Matrix(4,1) = Tensor[49];
-        Matrix(4,2) = Tensor[53];
-        Matrix(4,3) = 2.0*Tensor[46];
-        Matrix(4,4) = 0.0*Tensor[50];
-        Matrix(4,5) = 0.0*Tensor[51];
-
-        Matrix(5,0) = Tensor[54];
-        Matrix(5,1) = Tensor[58];
-        Matrix(5,2) = Tensor[62];
-        Matrix(5,3) = 2.0*Tensor[55];
-        Matrix(5,4) = 2.0*Tensor[59];
-        Matrix(5,5) = 2.0*Tensor[60];
-
+        if (Matrix.size1() != 6 || Matrix.size2() != 6) Matrix.resize(6, 6, false);
+
+        Matrix(0, 0) = Tensor[0];
+        Matrix(0, 1) = Tensor[4];
+        Matrix(0, 2) = Tensor[8];
+        Matrix(0, 3) = 2.0 * Tensor[1];
+        Matrix(0, 4) = 2.0 * Tensor[5];
+        Matrix(0, 5) = 2.0 * Tensor[6];
+
+        Matrix(1, 0) = Tensor[36];
+        Matrix(1, 1) = Tensor[40];
+        Matrix(1, 2) = Tensor[44];
+        Matrix(1, 3) = 2.0 * Tensor[37];
+        Matrix(1, 4) = 0.0 * Tensor[41];
+        Matrix(1, 5) = 0.0 * Tensor[42];
+
+        Matrix(2, 0) = Tensor[72];
+        Matrix(2, 1) = Tensor[76];
+        Matrix(2, 2) = Tensor[80];
+        Matrix(2, 3) = 2.0 * Tensor[73];
+        Matrix(2, 4) = 2.0 * Tensor[77];
+        Matrix(2, 5) = 2.0 * Tensor[78];
+
+        Matrix(3, 0) = Tensor[9];
+        Matrix(3, 1) = Tensor[13];
+        Matrix(3, 2) = Tensor[18];
+        Matrix(3, 3) = 2.0 * Tensor[10];
+        Matrix(3, 4) = 2.0 * Tensor[14];
+        Matrix(3, 5) = 2.0 * Tensor[15];
+
+        Matrix(4, 0) = Tensor[45];
+        Matrix(4, 1) = Tensor[49];
+        Matrix(4, 2) = Tensor[53];
+        Matrix(4, 3) = 2.0 * Tensor[46];
+        Matrix(4, 4) = 0.0 * Tensor[50];
+        Matrix(4, 5) = 0.0 * Tensor[51];
+
+        Matrix(5, 0) = Tensor[54];
+        Matrix(5, 1) = Tensor[58];
+        Matrix(5, 2) = Tensor[62];
+        Matrix(5, 3) = 2.0 * Tensor[55];
+        Matrix(5, 4) = 2.0 * Tensor[59];
+        Matrix(5, 5) = 2.0 * Tensor[60];
     }
+
     /**
-    * Generates the fourth order deviatoric unity tensor
-    * @param Unity the deviatoric unity (will be overwritten)
-    */
-    static void DeviatoricUnity(std::vector<std::vector<Matrix> >& Unity)
+     * Generates the fourth order deviatoric unity tensor
+     * @param Unity the deviatoric unity (will be overwritten)
+     */
+    static void DeviatoricUnity(std::vector<std::vector<Matrix>>& Unity)
     {
         Unity.resize(3);
 
-        Matrix kronecker(3,3);
-        noalias(kronecker)=ZeroMatrix(3,3);
-        for(unsigned int i=0; i<3; i++)
-        {
-            kronecker(i,i)=1;
+        Matrix kronecker(3, 3);
+        noalias(kronecker) = ZeroMatrix(3, 3);
+        for (unsigned int i = 0; i < 3; i++) {
+            kronecker(i, i) = 1;
```

Comment:
I think this is Identity matrix, no?

Reply:
Good catch, yes this could be simplified. I propose to do that separately to have this PR only be focused on formatting